src/_pytest/junitxml.py

```diff
@@ -27,10 +28,6 @@
 # Python 2.X and 3.X compatibility
 if sys.version_info[0] < 3:
     from codecs import open
```

Comment:
i believe it should be possible to go for `io.open` by now

Reply:
yeah I thought about this, but decided to leave it for another PR given `codecs.open` works more like python2.x open:

```pycon
>>> from codecs import open
>>> with open('f', 'w', encoding='utf-8') as f:
...     f.write(b'wat')
... 
>>> import io
>>> with io.open('f', 'w', encoding='utf-8') as f:
...     f.write(b'wat')
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TypeError: write() argument 1 must be unicode, not str
```

Reply:
most interesting, but also something we should unify towards the unicode only behaviour in junitxml